Make sapper flag --cwd for dev and build work when installation is in sub directory

floratmin opened the pull request:
This pull request solves the problem from https://github.com/sveltejs/sapper/issues/922 probably https://github.com/sveltejs/sapper/issues/1074 and some of the problems of https://github.com/sveltejs/sapper/issues/1575 .

With this commit it is possible to have the sapper installation in a sub directory and the build directory in a directory which can be located somewhere else.

The only thing to adjust in the project is the directory for loading the files from the static directory. For this we change the line to `sirv('./$cwd/static', { dev })` in the file `server.js` in the project created trough the sapper template.

The use of the `export_build` variable feels a little hacky to me, perhaps somebody finds a more elegant solution for this.

Many tests fail, but this fix works with a clean template and my sample template.
My sample template for testing the --cwd option can be found here: https://github.com/floratmin/sapper-subdirectory-example

benmccann commented on src/core/create_app.ts:
```diff
@@ -159,9 +163,11 @@ function generate_server_manifest(
 	dest: string,
-	dev: boolean
+	dev: boolean,
+	export_build: boolean
```
Why don't you just pass in the correct value for `root` (i.e. `cwd` or the `root` directory) instead of adding this variable to check if `root` should be overriden?

floratmin replied:
I thought that it would be more obvious what is happening to the path, than to morph `cwd` to `root` in `build.js` and using it as `root` in `create_app.ts`. If you prefer this approach I could change this. I could also update my sapper version to resolve the merge conflicts.

benmccann commented:
It looks like this PR has merge conflicts

benmccann commented on src/api/build.ts:
```diff
@@ -30,21 +31,24 @@ export async function build({
 
+	const root = path.resolve('.');
 	cwd = path.resolve(cwd);
 	src = path.resolve(cwd, src);
```
will this work? wouldn't it have to be relative to the root?

floratmin replied:
I need a reference to the root from where the build is executed. cwd can be supplied as an absolute or relative path. And furthermore in build.ts cwd is reassigned to its absolute path with `cwd = path.resolve(cwd);`. After this there is no possibility to get the root back. Instead of `path.resolve('.')` maybe using `process.cwd()` would be more clear. I chose the former because it is more in line with the handling of the other paths. An other option would be to use `path.relative(cwd, '.')`, to get the needed reference and then get the root with `path.resolve(cwd, path.relative(cwd, '.'));`. But i think we are here only turning in circles.

benmccann commented on src/api/build.ts:
```diff
@@ -30,21 +31,24 @@ export async function build({
 	output = 'src/node_modules/@sapper',
 	static: static_files = 'static',
 	dest = '__sapper__/build',
+	export_build = false,
```
I'm not particularly a fan of having this `export_build` flag to handle one type of build differently. Could we pass in the `root` instead?

floratmin replied:
I removed it and created an extra variable pointing to the root which gets passed to the build function.

floratmin commented:
I also updated my sample template at https://github.com/floratmin/sapper-subdirectory-example and added scripts with the cwd flag as absolute path. Every combination works. In my nx workspace I also used the additional flags --routes and --static with no problems.

benmccann commented:
The lint check and tests look like they're failing

floratmin commented:
@benmccann linting is passing now and from 150 tests only one is failing. To make the tests passing I had to made an adjustment in the sapper code. Otherwise I would have to edit 23 test files and change all occurrences of `build({ cwd: __dirname })` to `build({ cwd: __dirname, root: __dirname })`. I thought that also other people could use tests in this way and therefore preferred this approach.

benmccann commented:
Thanks. Sorry this got dropped a bit. All our focus has been on SvelteKit. It might be better to make sure anything you want to do is supported there. I'm a bit hesitant to make many changes to Sapper except for bug fixes

benmccann commented:
SvelteKit 1.0 is now out and Sapper is deprecated, so I'm going to close this
